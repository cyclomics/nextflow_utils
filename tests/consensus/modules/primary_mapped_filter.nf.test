nextflow_process {

    name "Test Process PrimaryMappedFilter"
    script "consensus/modules/samtools.nf"
    process "PrimaryMappedFilter"
    
    
    test("Filtering primary alignments") {

        when {
            process {
                """
                sample_id = "test_sample"
                bam_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.bam")
                bai_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.bam.bai")
                
                input[0] = Channel.of([sample_id, bam_file, bai_file])
                """
            }
        }
        then {
            assert process.success
            
            def bai_filename = "test_sample.primary_mapped.bam.bai"
            with(process.out[0]) {
                assert path(get(0)[1]).md5 == "a4df4798720072ca05d23674957c7f2e"
                assert path(get(0)[2]).getFileName().toString() == bai_filename
                assert path(get(0)[2]).size() > 0
            }
        }
    }
}
