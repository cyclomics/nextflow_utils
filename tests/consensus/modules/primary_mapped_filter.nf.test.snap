{
    "Filtering primary alignments": {
        "content": [
            {
                "0": [
                    [
                        "test_sample",
                        "test_sample.primary_mapped.bam:md5,a4df4798720072ca05d23674957c7f2e",
                        "test_sample.primary_mapped.bam.bai:md5,bffe1c97be07864c1acde1ca96c024df"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-10T16:54:39.119735"
    }
}